File: june/june13.py
def solve():
    t = int(input())
    for _ in range(t):
        n = int(input())
        k = int(input())
        a=[]
        for i in range(n):
            x=int(input())
            a.append(x)
        
        # Sort the card counts
        a.sort()
        
        # Start from the smallest value and try to equalize
        j=0
        mx=a[0]
        for i in range(n-1):
            d=a[i+1]-a[i]
            e=d*(i+1)
            if(e<=k):
                k-=e
                j=i+1
                mx=a[j]
            else:
                mx+=(k/(i+1))
                k%=(i+1)
                break;
        if(k>0):
            mx+(k/n)
            k%=n
        print(int(mx*n+k-j),'\n')
            
solve()
# Example usage:
# You can run the `solve()` function and provide input through standard input (stdin) to test the function.

File: june/june13.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

long long f(vector<long long>& a, long long k, int n) {
    long long m = a[0];
    int j = 0;
    for (int i = 0; i < n-1; ++i) {
        long long d = a[i+1] - a[i];
        long long e = (i+1) * d;
        if (k >= e) {
            k -= e;
            j = i+1;
            m = a[j];
        } else {
            m += k / (i+1);
            k %= (i+1);
            break;
        }
    }

    if (k > 0) {
        m += k / n;
        k %= n;
    }

    long long ans = m * n + (k - j);
    return ans;
}

void solve() {
    int n;
    long long k;
    cin >> n >> k;
    vector<long long> a(n);
    for (int i = 0; i < n; ++i) {
        cin >> a[i];
    }
    sort(a.begin(), a.end());
    cout << f(a, k, n) << endl;
}

int main() {
    int t;
    cin >> t;
    for (int i = 0; i < t; ++i) {
        solve();
    }
    return 0;
}
